solved the problem

FilippovichSergey opened the pull request:
solved the problem

FilippovichSergey committed:
solved the problem

FilippovichSergey committed:
updated Main class to match style

MarselYsup commented on src/main/java/school/faang/BJS2_72305/Army.java:
```diff
@@ -0,0 +1,40 @@
+            try {
+                thread.join();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
```
можно добавить   
```java
Thread.currentThread().interrupt(); 
```
 чтобы установить флаг прерывания текущего потока. Это позволяет потоку узнать, что его просят завершить работу или прервать текущую операцию.

MarselYsup commented on src/main/java/school/faang/BJS2_72305/Squad.java:
```diff
@@ -0,0 +1,26 @@
+@Getter
+@AllArgsConstructor
+public class Squad {
+    String name;
```
нарушение инкапсуляции - используй private

MarselYsup commented on src/main/java/school/faang/BJS2_72305/Squad.java:
```diff
@@ -0,0 +1,26 @@
+@AllArgsConstructor
+public class Squad {
+    String name;
+    List<Unit> squadMembers = new ArrayList<>();
```
используй final - чтобы никто не мог извне изменит лист,  также используй private - следуя принципу инкапсуляции

MarselYsup commented on src/main/java/school/faang/BJS2_72305/Army.java:
```diff
@@ -0,0 +1,40 @@
+
+public class Army {
+
+    List<Squad> squads = new ArrayList<>();
```
используй ```final``` - чтобы никто не мог извне изменит лист, также используй ```private``` - следуя принципу инкапсуляции

MarselYsup reviewed (commented):
Основная логика сделана отлично 👍
Поправь небольшие комменты

FilippovichSergey committed:
fixed according to comments:
class Squad. Changed the field squadMembers to private final
class Army. Changed the field squads to private final
